mindsdb/interfaces/datastore/datastore.py

```diff
@@ -40,8 +38,20 @@ def get_datasources(self):
         return datasource_arr
 
     def get_data(self, name, where=None, limit=None, offset=None):
-        # @TODO Apply filter directly to postgres/mysql/clickhouse/etc...  when the datasource is of that type
-        return get_sqlite_data(os.path.join(self.dir, name, 'sqlite.db'), where=where, limit=limit, offset=offset)
+        if offset is None:
+            offset = 0
+        ds = self.get_datasource_obj(name)
+
+        # @TODO Remove and add `offset` to the `filter` method of the datasource
+        filtered_ds = ds.filter(where=where, limit=limit+offset)
```

Comment:
if `limit=None` and `offset=None` then this line fail with:
```
unsupported operand type(s) for +: 'NoneType' and 'int'
```
limit arg should be `None` in this line if both is `None`